### PR TITLE
feat: rename pnpmfile.js to .pnpmfile.cjs

### LINKED ISSUE #1339: feature request: support .pnpmfile.js (dotfile)
Currently `pnpm` allows some advanced customization (i.e. hooks to fix third party packages) via the pnpmfile.js (see [FAQ](https://pnpm.js.org/docs/en/faq.html#solution-2)).

It seems like many tools use dot files to store in-repo config like this, so it's a little more obvious what is the repo's source code and what is a config file (example: `.eslintrc.js`). 

Could `pnpm` be modified to support, by default, `.pnpmfile.js` in addition to the current `pnpmfile.js`? 

I've seen that there's a way to change where the file is through the NPM config system, but it seems like a dotfile might be a more ideal default for most people -- especially since the .js extension could otherwise make people think it's part of the repository's source code rather than its config.

### Expected behavior:

pnpm looks for (hidden) `.pnpmfile.js` in addition to (not hidden)`pnpmfile.js`.

### Actual behavior:

pnpm only looks for `pnpmfile.js`.

### PR DESCRIPTION
**Pull request by zkochan**
close #1339

### CODE REVIEW

File: packages/pnpm/test/install/hooks.ts [hunk 1]
```diff
@@ -220,32 +220,32 @@ test('readPackage hook during update', async () => {
...
-test('fails when pnpmfile.js requires a non-existend module', async () => {
+test('fails when .pnpmfile.cjs requires a non-existend module', async () => {
   prepare()
 
-  await fs.writeFile('pnpmfile.js', 'module.exports = require("./this-does-node-exist")', 'utf8')
+  await fs.writeFile('.pnpmfile.cjs', 'module.exports = require("./this-does-node-exist")', 'utf8')
 
   const proc = execPnpmSync(['install', 'pkg-with-1-dep'])
 
   expect(proc.stdout.toString().includes('Error during pnpmfile execution')).toBeTruthy()
```
Comment (ExE-Boss) [inline packages/pnpm/test/install/hooks.ts]:
```suggestion
  expect(proc.stdout.toString()).toContain('Error during pnpmfile execution')
```

File: packages/pnpm/test/install/hooks.ts [hunk 2]
```diff
@@ -220,32 +220,32 @@ test('readPackage hook during update', async () => {
...
-test('prints meaningful error when there is syntax error in pnpmfile.js', async () => {
+test('prints meaningful error when there is syntax error in .pnpmfile.cjs', async () => {
   prepare()
 
-  await fs.writeFile('pnpmfile.js', '/boom', 'utf8')
+  await fs.writeFile('.pnpmfile.cjs', '/boom', 'utf8')
 
   const proc = execPnpmSync(['install', 'pkg-with-1-dep'])
 
   expect(proc.stderr.toString().includes('SyntaxError: Invalid regular expression: missing /')).toBeTruthy()
```
Comment (ExE-Boss) [inline packages/pnpm/test/install/hooks.ts]:
```suggestion
  expect(proc.stderr.toString()).toContain('SyntaxError: Invalid regular expression: missing /')
```

**Review by ExE-Boss — COMMENTED** [review]
Using [`expect(…).toContain(…)`](https://jestjs.io/docs/en/expect#tocontainitem) should make the CI failure easier to debug.

**Comment by zkochan** [comment]
This change might be not needed if this solution works: https://github.com/pnpm/pnpm/issues/3145#issuecomment-780246876

**Comment by zkochan** [comment]
I think it might make more sense to use .mjs instead of .cjs

**Pull request merged**